File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if s in t:
            return True
        if len(s)==0:
            return True
        if len(s)==1:
            if s not in t:
                return False
            else:
                return True
        n=len(s)
        m=len(t)
        k=0
        l=0
        count=0
        while(l<m):
            if s[k] not in t[l:len(t)]:
                return False
            if s[k]==t[l]:
                k+=1
                count+=1
                if count==n:
                    return True
            l+=1
        return False

        

File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if(s.size()>t.size()) return false;
        else if(s.size()==t.size()){
            if(s!=t) return false;
            return true;
        }
        else{
            int k=0;
            for(int i=0;i<t.size();i++){
                if(t[i]==s[k]) k++;
            }
            if(k>=s.size()) return true;
        }
        return false;
    }
};